internal/api/connectors/inventory/inventory.go

```diff
@@ -0,0 +1,175 @@
+				"fields[system_profile]": []string{"rhc_client_id", "owner_id"},
+			},
+		},
+	)
+
+	return &ApiHostGetHostSystemProfileByIdParams{
+		OrderBy:  &orderByParam,
+		OrderHow: &orderHowParam,
+		Fields:   &fields,
+	}
+}
+
+func NewInventoryClientWithHttpRequestDoer(cfg *viper.Viper, doer HttpRequestDoer) InventoryConnector {
+	client := &ClientWithResponses{
+		ClientInterface: &Client{
+			Server: fmt.Sprintf("%s://%s:%d%s", cfg.GetString("inventory.connector.scheme"), cfg.GetString("inventory.connector.host"), cfg.GetInt("inventory.connector.port"), basePath),
+			Client: utils.NewMeasuredHttpRequestDoer(doer, "inventory", "GetHostConnectionDetails"),
+			RequestEditor: func(ctx context.Context, req *http.Request) error {
+				req.Header.Set(constants.HeaderRequestId, request_id.GetReqID(ctx))
+
+				if identity, ok := ctx.Value(constants.HeaderIdentity).(string); ok {
+					req.Header.Set(constants.HeaderIdentity, identity)
+				}
+
+				return nil
+			},
+		},
+	}
+
+	return &inventoryConnectorImpl{
+		client: client,
+	}
+}
+
+func NewInventoryClient(cfg *viper.Viper) InventoryConnector {
+	httpClient := http.Client{
+		Timeout: time.Duration(cfg.GetInt64("inventory.connector.timeout") * int64(time.Second)),
+	}
+
+	return NewInventoryClientWithHttpRequestDoer(cfg, &httpClient)
+}
+
+func (this *inventoryConnectorImpl) getHostDetails(
+	ctx context.Context,
+	IDs []string,
+	orderBy string,
+	orderHow string,
+	limit int,
+	offset int,
+) (details []HostOut, err error) {
+
+	params := createHostGetHostByIdParams(orderBy, orderHow)
+
+	response, err := this.client.ApiHostGetHostByIdWithResponse(ctx, IDs, params)
+
+	if err != nil {
+		return nil, err
+	}
+
+	if response.JSON200 == nil {
+		return nil, utils.UnexpectedResponse(response.HTTPResponse)
+	}
+
+	return response.JSON200.Results, err
+}
+
+func (this *inventoryConnectorImpl) getSystemProfileDetails(
+	ctx context.Context,
+	IDs []string,
+	orderBy string,
+	orderHow string,
+	limit int,
+	offset int,
+) (details []HostSystemProfileOut, err error) {
+
+	params := createHostGetHostSystemProfileByIdParams(orderBy, orderHow)
+
+	response, err := this.client.ApiHostGetHostSystemProfileByIdWithResponse(ctx, IDs, params)
+
+	if err != nil {
+		return nil, err
+	}
+
+	if response.JSON200 == nil {
+		return nil, utils.UnexpectedResponse(response.HTTPResponse)
+	}
+
+	return response.JSON200.Results, nil
+}
+
+func (this *inventoryConnectorImpl) GetHostConnectionDetails(ctx context.Context, IDs []string, order_how string, order_by string, limit int, offset int) (details []HostDetails, err error) {
+
+	hostResults, err := this.getHostDetails(ctx, IDs, order_how, order_by, limit, offset)
+
+	if err != nil {
+		return nil, err
+	}
+
+	systemProfileResults, err := this.getSystemProfileDetails(ctx, IDs, order_by, order_how, limit, offset)
+
+	if err != nil {
+		return nil, err
+	}
+
+	hostConnectionDetails := make([]HostDetails, len(IDs))
+	for i, host := range hostResults {
```

Comment:
We might need to be careful here.  I think we are assuming that the inventory methods are returning the same ids in the same order from the separate methods.  I think we need to make sure that the ids are the same when we are constructing the HostDetails struct below.

For example, imagine a host we are looking for is deleted after the call to getHostDetails() but before getSystemProfileDetails().  I think the order of the system profile results will not match so the data in the HostDetails{} could be mismatched.

I guess it really depends on how inventory responds.  I might be overly paranoid, I'm not sure.   But I might error on the side of caution and make sure the ids line up between the two calls.  In order to help with that, I might make the getSystemProfileDetails() return a map of system profiles keyed by the id.  Just something to think about.

Reply:
@dehort I believe that with both methods the passed in "order_by" and "order_how" parameters will make sure that the returning ids are in the same order, but I could also check the ids just to make sure!